### PR TITLE
Fix JS error on the create account page with multishipping

### LINKED ISSUE #32629: Product count not appear on Mini cart
Product count not appear on Mini cart 

### Preconditions (*)

1. Magento 2.4

### Steps to reproduce (*)
- As guest user add 2-3 products to cart 
- Shopping Cart 
- Click on Checkout with multiple addresses 
- Login page -> Click on Create an account button 
- Observe at mini cart 

### Expected result (*)
Product count should be appear on mini cart 

### Actual result (*)
https://prnt.sc/y4ut7w

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### LINKED ISSUE #32630: Password strength not updating issue on register page
### Preconditions (*)

1. Magento 2.4

### Steps to reproduce (*)

- As guest user add 2-3 products to cart 
- Shopping Cart 
- Click on Checkout with multiple addresses 
- Login page -> Click on Create an account button 
- Create an account page -> Enter value in password 
- Observe password strength not update

### Expected result (*)

Password strength should be updated as enter values 

### Actual result (*)

https://prnt.sc/y4x64i

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### LINKED ISSUE #32632: Page refresh on subscribe for footer newsletter section
Page refresh on subscribe for footer newsletter section

### Preconditions (*)

1. Magento 2.4
2.

### Steps to reproduce (*)

- As guest user add 2-3 products to cart 
- Shopping Cart 
- Click on Checkout with multiple addresses 
- Login page -> Click on Create an account button 
- Create an account page -> Footer -> Newsletter section
- Without enter email click on Subscribe button
- 
### Expected result (*)

Require validation should appear below email text field

### Actual result (*)

Page refresh and no validation message appear

---
Please provide [Severity](https://devdocs.magento.com/guides/v2.4/contributor-guide/contributing.html#backlog) assessment for the Issue as Reporter. This information will help during Confirmation and Issue triage processes.

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [ ] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [ ] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

### PR DESCRIPTION
**Pull request by ihor-sviziev**
### Description (*)
My PR fixes JS error that happened on the Create account page using multishipping checkout.
URL looks like this https://example.com/multishipping/checkout/addresses/

Covered in: https://github.com/magento/magento2/pull/32638

### Fixed Issues (if relevant)

1. Fixes https://github.com/magento/magento2/issues/32629
2. Fixes https://github.com/magento/magento2/issues/32630
3. Fixes https://github.com/magento/magento2/issues/32632

### Manual testing scenarios (*)

1. As guest user add 2-3 products to cart
2. Shopping Cart
3. Click on Checkout with multiple addresses
4. Login page -> Click on Create an account button

**Expected result:**
There shouldn't be any JS errors

**Actual result:**
We have a JS error that causing multiple issues, listed in "fixed issues" section:
image
image

It was caused by an incorrect JSON file - in the source code we see the following code:
image

**Note:** I don't think that MFTF tests will be effective here, as you still CAN create an account if you'll put all needed info correctly, while some JS part is broken on the page.

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by ihor-sviziev** [comment]
@magento run all test

**Comment by ihor-sviziev** [comment]
@magento run all tests

File: app/code/Magento/Customer/view/frontend/templates/form/register.phtml [hunk 1]
```diff
@@ -355,9 +355,9 @@ script;
                 "regionInputId": "#region",
                 "postcodeId": "#zip",
                 "form": "#form-validate",
-                "regionJson": {$regionJson},
-                "defaultRegion": "{$regionId}",
-                "countriesWithOptionalZip": {$countriesWithOptionalZip}
+                "regionJson": <?= /* @noEscape */ $regionJson,
```
Comment (srsathish92) [inline app/code/Magento/Customer/view/frontend/templates/form/register.phtml]: @ihor-sviziev Seems missing "?>" close tag. Can you please check

Comment (ihor-sviziev) [reply app/code/Magento/Customer/view/frontend/templates/form/register.phtml]:
```suggestion
                "regionJson": <?= /* @noEscape */ $regionJson ?>,
```

**Comment by ihor-sviziev** [comment]
@magento run all tests

**Comment by eduard13** [comment]
Hi @ihor-sviziev, in my opinion this issue should have at least a `P2`. Also I guess that we should be able to cover this case with a MFTF test, that will check the minicart on the mentioned page.

@magento give me 2.4-develop instance

**Comment by ihor-sviziev** [comment]
@eduard13 unfrortunately, I don't have enough knowledge about MFTF. Could you help me with that?

**Comment by ihor-sviziev** [comment]
@magento run Functional Tests B2B, WebAPI Tests

**Comment by eduard13** [comment]
ℹ️ The test coverage is provided in scope of #32638.

**Comment by gabrieldagama** [comment]
@magento run all tests

**Comment by eduard13** [comment]
@magento run Unit Tests, Functional Tests EE, Functional Tests B2B

**Comment by hostep** [comment]
@ihor-sviziev, I think the following issues can be added to the list of issues to be fixed:
- https://github.com/magento/magento2/issues/30752
- https://github.com/magento/magento2/issues/30099
- https://github.com/magento/magento2/issues/32128
- ...

There is even another PR trying to fix it https://github.com/magento/magento2/pull/27608 (even though that one fixes some more issues at the same time I think)

Just FYI

**Comment by ihor-sviziev** [comment]
@hostep tyxfr the list. Actually I think it's better to fix the JS error and not any other improvements right now. They might be added later if needed

**Comment by eduard13** [comment]
@magento run Functional Tests EE

**Review by eduard13 — APPROVED** [review]
The failed test doesn't seem related to these changes.

**Comment by magento-engcom-team** [comment]
Hi @eduard13, thank you for the review.
[ENGCOM-8978](https://jira.corp.magento.com/browse/ENGCOM-8978) has been created to process this Pull Request

**Comment by engcom-Oscar** [comment]
:heavy_check_mark: QA passed

Was checked case from Manual testing scenarios 

:heavy_check_mark: **After:**
Screenshot from 2021-04-01 13-29-52
Screenshot from 2021-04-01 13-29-07

**Comment by ihor-sviziev** [comment]
Hi,

The issue was already fixed in https://github.com/magento/magento2/commit/461509df6db50edf65de181ecb7decc27519a748. I'm closing this PR as it's not relevant anymore

**Pull request closed without merging**